### ISSUE TITLE
ProtoBuf fails to deserialize instances that have an empty list as its field

### ISSUE THREAD

**Issue by qradimir** [Kotlin/kotlinx.serialization#1254]
**Describe the bug**

ProtoBuf fails to deserialize instances that have an empty list as its field.

**To Reproduce**

```kotlin
@Serializable
data class Foo(val values: List<Long>)

@Serializable
data class Bar(val foo: Foo)

@OptIn(ExperimentalSerializationApi::class)
class Test {

    @Test
    fun test() {
        val expected = Bar(Foo(emptyList()))
        val serialized = ProtoBuf.encodeToByteArray(Bar.serializer(), expected)
        val actual = ProtoBuf.decodeFromByteArray(Bar.serializer(), serialized)
        assertEquals(expected, actual)
    }
}
```

Fails with 
```
Field 'values' is required, but it was missing
kotlinx.serialization.MissingFieldException: Field 'values' is required, but it was missing
	at Foo.<init>(Test.kt)
	at Foo$$serializer.deserialize(Test.kt)
	at Foo$$serializer.deserialize(Test.kt:9)
	at kotlinx.serialization.protobuf.internal.ProtobufDecoder.decodeSerializableValue(ProtobufDecoding.kt:175)
	at kotlinx.serialization.protobuf.internal.ProtobufTaggedDecoder.decodeSerializableElement(ProtobufTaggedDecoder.kt:80)
	at Bar$$serializer.deserialize(Test.kt)
	at Bar$$serializer.deserialize(Test.kt:12)
	at kotlinx.serialization.protobuf.internal.ProtobufDecoder.decodeSerializableValue(ProtobufDecoding.kt:175)
	at kotlinx.serialization.protobuf.internal.ProtobufDecoder.decodeSerializableValue(ProtobufDecoding.kt:165)
	at kotlinx.serialization.protobuf.ProtoBuf.decodeFromByteArray(ProtoBuf.kt:137)
	at Test.test(Test.kt:22)
```

**Expected behavior**

Runs without failures.

**Environment**
 - Kotlin version: 1.4.10
 - Library version: 1.0.1
 - Kotlin platforms: JVM
 - Gradle version:  6.6.1

**Comment by qradimir** [comment]
The workaround is to set a default value `emptyList()` for the property.

**Comment by sandwwraith** [comment]
Probably duplicate of #806?

**Comment by qradimir** [comment]
@sandwwraith Seems so.

**Comment by shanshin** [comment]
Fixed in 1.2.2 with 8f82c6b
